src/index.ts

```diff
@@ -156,8 +157,9 @@ function compression<T, A extends Algorithm>(opts: ViteCompressionPluginConfig<T
       const handle = async (file: string, meta: CompressMetaInfo) => {
         if (meta.effect) return
         const bundle = bundles[file]
-        const source = bundle.type === 'asset' ? bundle.source : bundle.code
-        const compressed = await transfer(Buffer.from(source), zlib.algorithm, zlib.options)
+        const source = Buffer.from(bundle.type === 'asset' ? bundle.source : bundle.code)
+        const compressed = await transfer(source, zlib.algorithm, zlib.options)
+        if (skipIfLargerOrEqual && len(compressed) >= len(source)) return

```

Comment:
I can't know you mean? Why do we need to add this logic here? Because in the previous logic. we skip the file that size less than the original file ( So it' will be work 
I guess maybe because some files will be larger after compression so we need to skip?

Reply:
Oh. my mistake. I ignored your description. you're right

Reply:
The file size check that you mentioned [here](https://github.com/nonzzz/vite-plugin-compression/blob/master/src/index.ts#L119-L124) only checks if the original file is larger than a certain threshold. This is something different.

Reply:
The feature added in this PR also checks if the compressed output is smaller than the original content (and ignores processing the compressed result if it is not needed).